src/Nodes1DProvisioner.cpp

```diff
@@ -159,61 +160,43 @@ namespace blitzdg {
     }
 
     void Nodes1DProvisioner::buildConnectivityMatrices() {
-        firstIndex ii;
-        secondIndex jj;
-        thirdIndex kk;
-
-        index_type totalFaces = NumFaces*NumElements;
+        index_type totalFaces = NumFaces * NumElements;
         index_type numVertices = NumElements + 1;
+        index_type localVertNum[2] = {0, 1};
 
-        index_type localVertNum[2];
-        localVertNum[0] = 0; localVertNum[1] = 1;
-
-        // Build global face-to-vertex array. (should be sparse matrix in 2D/3D).
-        real_matrix_type FToV(totalFaces, numVertices);
-        FToV = 0*jj;
-
-        index_matrix_type & E2V = *EToV;
-
+        // Build global face-to-vertex array. Note: we actually
+        // build its transpose for easy column access.
+        CSCMat FToVtrans(numVertices, totalFaces, totalFaces);
+        const index_matrix_type& E2V = *EToV;
         index_type globalFaceNum = 0;
-        for (index_type k=0; k < NumElements; k++) {
-            for (index_type f=0; f < NumFaces; f++) {
+        for (index_type k = 0; k < NumElements; ++k) {
```

Comment:
Any reason why we're straying from the convention of using k++ here? I'd at least like to see this done consistently throughout the project in for-statements.

Reply:
It's better to use ++k as it is slightly more efficient - it doesn't have to store and return the old value. Only use k++ if you need the value before incrementing, otherwise, always use ++k.

Reply:
we should make it a point to fix this elsewhere if that's the case.

Reply:
Good call.